sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryOptions.java

```diff
@@ -78,4 +79,11 @@
   Integer getLatencyLoggingFrequency();
 
   void setLatencyLoggingFrequency(Integer value);
+
+  @Experimental
+  @Description("Whether dynamic sharding is enabled for writing to BigQuery in streaming.")
+  @Default.Boolean(false)
+  Boolean getEnableStreamingAutoSharding();
```

Comment:
Done.

Reply:
I was thinking that adding the option to BigQueryOptions might be easier for the users to switch on/off the feature. If it is an option of BigQueryIO users would need to update their code (vs. adding a flag) right? It might not be a concern...